sdk/core/azure-core/tests/test_request_id_policy.py

```diff
@@ -6,23 +6,41 @@
 from azure.core.pipeline.policies import RequestIdPolicy
 from azure.core.pipeline.transport import HttpRequest
 from azure.core.pipeline import PipelineRequest, PipelineContext
+from mock import patch
+from itertools import product
 import pytest
 
-def test_request_id_policy_fix_id():
-    """Test policy with no other policy and happy path"""
-    test_request_id = 'test_request_id'
-    request_id_policy = RequestIdPolicy(request_id=test_request_id)
-    request = HttpRequest('GET', 'http://127.0.0.1/')
-    pipeline_request = PipelineRequest(request, PipelineContext(None))
-    request_id_policy.on_request(pipeline_request)
-    assert request.headers["x-ms-client-request-id"] == test_request_id
+auto_request_id_values = (True, False, None)
+request_id_init_values = ("foo", None)
```

Comment:
As you correctly noticed, None and not provided are two differents scenarios, so you need an "_unset" here too. Then this:
``` python
    if request_id_init is not None:
        kwargs['request_id'] = request_id_init
```
becomes this:
``` python
    if request_id_init is not _Unset:
        kwargs['request_id'] = request_id_init
```
and then you actually test the three interesting value: random string, None, not set

Reply:
Same applies to all way to provide any input parameter